src/data/arena-tag.ts

```diff
@@ -989,9 +975,15 @@ class StealthRockTag extends ArenaTrapTag {
         return true;
       }
       const damage = toDmgValue(pokemon.getMaxHp() * damageHpRatio);
-      globalScene.queueMessage(
-        i18next.t("arenaTag:stealthRockActivateTrap", { pokemonNameWithAffix: getPokemonNameWithAffix(pokemon) }),
-      );
+      if (this.damagingType === Type.ROCK) {
+        globalScene.queueMessage(
+          i18next.t("arenaTag:stealthRockActivateTrap", { pokemonNameWithAffix: getPokemonNameWithAffix(pokemon) }),
+        );
+      } else {
+        globalScene.queueMessage(
+          i18next.t("arenaTag:sharpSteelActivateTrap", { pokemonNameWithAffix: getPokemonNameWithAffix(pokemon) }),
+        );
+      }
```

Comment:
You probably want to avoid hard-coding those messages and instead provide the message as parameter.